### PR TITLE
Add AFK Command and Listener

### LINKED ISSUE #8: AFK
Auto-set AFK status when the player is inactive. And kick after the time allowed has expired.

Make kicking OPs configurable.

### PR DESCRIPTION
**Pull request by Aiden880**
9th times the charm. Fixes #8

### CODE REVIEW

File: src/main/java/com/helion3/bedrock/Bedrock.java [hunk 1]
```diff
@@ -61,6 +65,7 @@ public void onServerStart(GameStartedServerEvent event) {
         parentDirectory = defaultConfig.getParentFile();
 
         // Commands
+        game.getCommandManager().register(this, BedrockCommands.getCommand(), "afk");
```
Comment (viveleroi) [inline src/main/java/com/helion3/bedrock/Bedrock.java]: This is using the wrong class for the command.

File: src/main/java/com/helion3/bedrock/Bedrock.java [hunk 2]
```diff
@@ -47,6 +50,7 @@
     private static final MessageManager messageManager = new MessageManager();
     private static File parentDirectory;
     private static final PlayerConfigManager playerConfigManager = new PlayerConfigManager();
+    public static List<Player> afkPlayers = new ArrayList<Player>();
```
Comment (viveleroi) [inline src/main/java/com/helion3/bedrock/Bedrock.java]: I've decided that we'll group related methods/lists like these in "manager" classes. So this list, and the add/remove afk methods will be moved to an `AfkManager` class. You can see my existing examples for `MessageManager` and `PlayerConfigManager`

File: src/main/java/com/helion3/bedrock/commands/AfkCommand.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+/**
+ * Created by Aiden on 21/02/2016.
+ */
+public class AfkCommand {
+    private AfkCommand() {}
+
+    public static CommandSpec getCommand() {
+        return CommandSpec.builder()
+                .arguments(
```
Comment (viveleroi) [inline src/main/java/com/helion3/bedrock/commands/AfkCommand.java]: This whole block is indented twice too far, and the following line is indent once too far.

File: src/main/java/com/helion3/bedrock/Bedrock.java (hunk 1 above)
Comment (Aiden880) [inline src/main/java/com/helion3/bedrock/Bedrock.java]: Fixed

File: src/main/java/com/helion3/bedrock/commands/AfkCommand.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+                        GenericArguments.playerOrSource(Text.of("player"))
+                )
+                .description(Text.of("Toggle player as AFK."))
+                .permission("bedrock.afk")
+                .executor((source, args) -> {
+
```
Comment (viveleroi) [inline src/main/java/com/helion3/bedrock/commands/AfkCommand.java]: No empty new line here please.

File: src/main/java/com/helion3/bedrock/Bedrock.java (hunk 2 above)
Comment (Aiden880) [inline src/main/java/com/helion3/bedrock/Bedrock.java]: Me and you are going to fall out

File: src/main/java/com/helion3/bedrock/commands/AfkCommand.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+                    // Define player as the source, confirmed instance of Player
+                    Player player = (Player) source;
+
+                    // Set user as AFK.
+                    // @TODO: 21/02/2016 Remember to add logic to kick player after x time
+                    // Add user to afklist
+                    Bedrock.afkPlayers.add(player);
+
+                    // Send message to "TO_ALL" message channel, build the text
+                    MessageChannel.TO_ALL.send(Text.builder(player.getName() + " is now AFK.").build());
```
Comment (viveleroi) [inline src/main/java/com/helion3/bedrock/commands/AfkCommand.java]: You can use `Text.of(player.getName() + " is now AFK.")` instead of the builder.

Comment (Aiden880) [inline src/main/java/com/helion3/bedrock/commands/AfkCommand.java]: Using text.builder for future formatting - assuming we won't leave it plain white?

Comment (viveleroi) [inline src/main/java/com/helion3/bedrock/commands/AfkCommand.java]: Text.of accepts colors and styles, so we should be fine. Builders are better when we can't do it all in one line.

File: src/main/java/com/helion3/bedrock/listeners/JoinListener.java [hunk 6]
```diff
@@ -24,12 +24,29 @@
...
 public class JoinListener {
     @Listener
     public void onPlayerJoin(final ClientConnectionEvent.Join event) {
-        Bedrock.getPlayerConfigManager().loadPlayer(event.getTargetEntity());
+        Player player = event.getTargetEntity();
+        Bedrock.getPlayerConfigManager().loadPlayer(player);
+        if(AfkManager.afkPlayers.contains(player)){
+            AfkManager.afkPlayers.remove(player);
+        }
+        if(player.hasPermission("bedrock.fly.join")) {
```
Comment (viveleroi) [inline src/main/java/com/helion3/bedrock/listeners/JoinListener.java]: Can you remove this for now, this should be handled by sponge and if it's not, we need a bug report.

**Comment by viveleroi** [comment]
Merged as part of 6484525da84e8716cb23d240c6189f937cee3fb2

**Pull request closed without merging**